src/Operation/ChangeStream.php

```diff
@@ -0,0 +1,159 @@
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnexpectedValueException;
+use MongoDB\Exception\UnsupportedException;
+use stdClass;
+use Traversable;
+
+/**
+ * Operation for the changeStream command.
+ *
+ * @api
+ * @see \MongoDB\Collection::changeStream()
+ * @see http://docs.mongodb.org/manual/reference/command/changeStream/
+ */
+class ChangeStream implements Executable
+{
+    private $databaseName;
+    private $collectionName;
+    private $resumeToken;
+    private $pipeline;
+    private $readPreference;
+    private $options;
+
+    /**
+     * Constructs a changeStream command.
+     *
+     * Supported options:
+     *
+     *  * fullDocument (string): Allowed values: ‘default’, ‘updateLookup’.
+     *    Defaults to ‘default’.  When set to ‘updateLookup’, the change
+     *    notification for partial updates will include both a delta describing
+     *    the changes to the document, as well as a copy of the entire document
+     *    that was changed from some time after the change occurred. For forward
+     *    compatibility, a driver MUST NOT raise an error when a user provides
+     *    an unknown value. The driver relies on the server to validate this
+     *    option.
+     *
+     *  * resumeAfter (document): Specifies the logical starting point for the
+     *    new change stream.
+     *
+     *  * maxAwaitTimeMS (integer): The maximum amount of time for the server to
+     *    wait on new documents to satisfy a change stream query.
+     *
+     *  * batchSize (integer): The number of documents to return per batch.
+     *
+     *    This option is sent only if the caller explicitly provides a value.
+     *    The default is to not send a value.
+     *
+     *  * collation (document): Specifies a collation.
+     *
+     *    This option is sent only if the caller explicitly provides a value.
+     *    The default is to not send a value.
+     *
+     * @param string         $databaseName   Database name
+     * @param string         $collectionName Collection name
+     * @param array          $pipeline       List of pipeline operations
+     * @param array          $options        Command options
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     */
+    public function __construct($databaseName, $collectionName, array $pipeline, array $options = [])
+    {
+        $expectedIndex = 0;
+
+        foreach ($pipeline as $i => $operation) {
+            if ($i !== $expectedIndex) {
+                throw new InvalidArgumentException(sprintf('$pipeline is not a list (unexpected index: "%s")', $i));
+            }
+
+            if ( ! is_array($operation) && ! is_object($operation)) {
+                throw InvalidArgumentException::invalidType(sprintf('$pipeline[%d]', $i), $operation, 'array or object');
+            }
+
+            $expectedIndex += 1;
+        }
```

Comment:
Aggregate will already validate the `$pipeline`, so I think we can do without this here and just trust `array_unshift()` to prepend our new `$changeStream` pipeline operator to the array. Based on some local testing, it does the sensible thing.

**Sequential numerically-indexed array**

```
>>> $a = [0=>'a',1=>'b']
=> [
     "a",
     "b",
   ]
>>> array_unshift($a, 'foo')
=> 3
>>> $a
=> [
     "foo",
     "a",
     "b",
   ]
```

**Associative array**

```
>>> $a = ['foo'=>'a',1=>'b']
=> [
     "foo" => "a",
     1 => "b",
   ]
>>> array_unshift($a, 'hello')
=> 3
>>> $a
=> [
     0 => "hello",
     "foo" => "a",
     1 => "b",
   ]
```

**Non-sequential numerically-indexed array**

```
>>> $a = [-1=>'a',1=>'b']
=> [
     -1 => "a",
     1 => "b",
   ]
>>> array_unshift($a, 'foo')
=> 3
>>> $a
=> [
     "foo",
     "a",
     "b",
   ]
```

This last example actually turns an otherwise invalid pipeline into a valid one (PHP appears to re-index the array), but I don't think that's a problem.